frontend/src/components/Preview.vue

```diff
@@ -0,0 +1,82 @@
+<template>
+  <v-container>
+    <v-row align="center" justify="center">
+      <v-col>
+        <v-row justify="center">
+          <v-container class="preview-container">
+            <router-link to="/">
+              <v-img class="preview-img" :src="preview" contain/>
+            </router-link>
+          </v-container>
+        </v-row>
+      </v-col>
+      <v-col>
+        <v-container class="preview-container">
+          <v-list two-line>
+            <template v-for="(item, index) in items">
+              <v-list-item class="resource-list" :href="item.link" @mouseover="preview = item.src" :key="item.title">
+                <v-list-item-content>
+                  <div class="text-h4" v-text="item.title" />
+                  <div class="text-subtitle-1" v-text="item.description" />
+                </v-list-item-content>
+              </v-list-item>
+              <v-divider v-if="index != items.length-1" :key="index" />
+            </template>
+          </v-list>
+          <div class="button-container"> 
+            <router-link to="/resources">
+              <v-btn text>All resources ></v-btn>
+            </router-link>
+          </div>
+
+        </v-container>
+      </v-col>
+    </v-row>
+  </v-container>
+</template>
+
+<script>
+import APIClient from '../utils/APIClient';
+
+export default {
+  name: 'Preview',
+  data: () => ({
+    items: [],
+    preview: "",
+  }),
+  mounted() {
+    APIClient.resourcesAPI('/preview')
+      .then((responseJson) => {
+        this.items = responseJson;
+        this.preview = responseJson[0].src
```

Comment:
What if this fetch fails. I think you should be checking (possibly http status code) before indexing `[0]`. I'd go ahead and put a v-if around the outermost v-container; if this fetch fails (either due to connection errors or a database error), there's no reason to render the component.

Looking at other components of the website, I don't think we are checking error codes enough (I'm guilty of this too🙄).

Reply:
Yeah that's what I'm afraid of too. I need to go through and have a more rigorous testing framework for the frontend. I'll close the request for now and get back to you.

Reply:
@glebme ohh I was going to say go ahead and merge this PR. Testing is outside the scope of this ticket, that can be it's own PR.

Reply:
No its ok. I had a rethink on how its designed and tested. I'm trying to implement that resources so you can review all in one go.